lib/modules/manager/bazel-module/context.ts

```diff
@@ -0,0 +1,164 @@
+import type {
+  AllFragments,
+  ArrayFragment,
+  ChildFragments,
+  RecordFragment,
+} from './fragments';
+import * as fragments from './fragments';
+
+// Represents the fields that the context must have.
+export interface CtxCompatible {
+  results: RecordFragment[];
+  stack: AllFragments[];
+}
+
+export class CtxProcessingError extends Error {
+  readonly current: AllFragments;
+  readonly parent?: AllFragments;
+  constructor(current: AllFragments, parent?: AllFragments) {
+    const msg = `Invalid context state. current: ${current.type}, parent: ${
+      parent?.type ?? 'none'
+    }`;
+    super(msg);
+    this.name = 'CtxProcessingError';
+    this.current = current;
+    this.parent = parent;
+  }
+}
+
+export class Ctx implements CtxCompatible {
+  results: RecordFragment[];
+  stack: AllFragments[];
+
+  constructor(results: RecordFragment[] = [], stack: AllFragments[] = []) {
```

Comment:
```suggestion
  private constructor(results: RecordFragment[], stack: AllFragments[]) {
```
so it's not used outside of this class and I think we don't need default values here.

Reply:
If I make it private, I won't be able to instantiate it in the test file. Correct?

Reply:
@viceice Could I leave the constructor as is so that I can execute the tests that I have for it?

Reply:
@viceice The `Context.as()` function no longer exists. I was able to remove it with the update to `good-enough-parser`.